core/drivers/stm32_rtc.c

```diff
@@ -778,7 +1073,28 @@ static TEE_Result stm32_rtc_probe(const void *fdt, int node,
 	res = stm32_rtc_init();
 	if (res)
 		return res;
-	rtc_register(&stm32_rtc);
+
+	if (IS_ENABLED(CFG_DRIVERS_RTC))
+		rtc_register(&stm32_rtc);
+
+	if (IS_ENABLED(CFG_CORE_ASYNC_NOTIF) && rtc_dev.is_secured) {
```

Comment:
Async notif is needed only for the RTC PTA service: add `IS_ENABLED(CFG_RTC_PTA)`.